File: 5a.php
<?php

$fname = "5.dat";
$fin = fopen($fname, "r");
$data = fread($fin, filesize($fname));
$lines = explode("\n", $data);
$vowels = "/[aeiou]/";
$dblletters = "/(.)\\1/";
$nice = 0;
for ($i=0; $i<count($lines)-1; $i++) {
    preg_match_all($vowels, $lines[$i], $matches);
    $vows = count($matches[0]);
    $dbl = preg_match($dblletters, $lines[$i]);
    $req3 =  strpos($lines[$i], "ab") !== false ||
             strpos($lines[$i], "cd") !== false ||
             strpos($lines[$i], "pq") !== false ||
             strpos($lines[$i], "xy") !== false;
    echo $vows . ' ' . $dbl . ' ' . ($req3?"True":"False") . PHP_EOL;
    if (($vows > 2) && $dbl && !$req3) {
        $nice++;
    }
}
echo $nice.PHP_EOL;
fclose($fin);

?>

File: 5a.py
#!/usr/bin/python3
import re

with open('5.dat') as f:
    data = f.read()
lines = data.split('\n')
lines.pop()
nice = 0
for word in lines:
    vows = len(re.findall('[aeiou]', word))
    dbl = re.search(r"(.)\1", word)
    req3 = False
    for x in ["ab", "cd", "pq", "xy"]:
        req3 = req3 or x in word
    if dbl is None:
        print(vows, 0, req3)
    else:
        print(vows, len(dbl.groups()), req3)
    if vows > 2 and dbl and not req3:
        nice += 1
print(nice)
